### PR TITLE
[hpx] Update hpx to V1.6.0

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

**Comment by PhoebeHui** [comment]
The gsoap failures would be fixed by PR https://github.com/microsoft/vcpkg/pull/16270.

**Comment by hkaiser** [comment]
> The gsoap failures would be fixed by PR #16270.

Not sure what we can do about this, moreover as HPX does not depend on gsoap.

**Comment by PhoebeHui** [comment]
@hkaiser, gsoap triggerred here due to its dependency rely on hpx, no changes required in this PR, after the baseline issue fixed, I will rerun this PR. thanks for your contribution!

File: ports/hpx/CONTROL [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Source: hpx
-Version: 1.5.1
+Version: 1.6.0
 Port-Version: 1
```
Comment (PhoebeHui) [inline ports/hpx/CONTROL:3]:
```suggestion
Port-Version: 0
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@hkaiser, thanks for the PR!

Could you update the port version and versions as following steps?
1. Update the port version, 
2. Submmit the changes.
3. Undo the changes in versions/h-/hpx.json.
4. Execute './vcpkg x-add-version hpx'.
5. Submmit the changes.

**Comment by hkaiser** [comment]
@PhoebeHui better now?

**Comment by PhoebeHui** [comment]
@hkaiser, in order to support new feature versioning, we need submmit a changes only for updating the versions after all changes for the ports has been committed.

Please execute './vcpkg x-add-version --overwrite-version hpx' and commit the changes.

```
Error: While reading versions for port hpx from file: C:\a\1\s\versions\h-\hpx.json
       File declares version `1.6.0` with SHA: 3e07ee0cdfdf093d41ddb40267f5910f71bfd944
       But local port with the same verion has a different SHA: 48f07b4410f57c083f6f17aa1d79addc533c3772
       Please update the port's version fields and then run:

           vcpkg x-add-version hpx

       to add a new version.
```

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

**Comment by ras0219-msft** [comment]
Thanks for the update PR!

**Pull request merged**